root/io/json/test_classes.h

```diff
@@ -0,0 +1,852 @@
+      TJsonEx11() : TJsonEx1() {
+         fInt2 = 22;
+      }
+};
+
+// _______________________________________________________________
+
+class TJsonEx2 {
+   protected:
+     int        fTest1[4];
+     int        fTest2[2][2];
+     bool       fBool[2][3][4];
+     char       fChar[2][3][4];
+     short      fShort[2][3][4];
+     int        fInt[2][3][4];
+     long       fLong[2][3][4];
+     float      fFloat[2][3][4];
+     double     fDouble[2][3][4];
+   public:
+     TJsonEx2()
+     {
+        fTest1[0] = 11111;
+        fTest1[1] = 22222;
+        fTest1[2] = 33333;
+        fTest1[3] = 44444;
+
+        fTest2[0][0] = 1;
+        fTest2[0][1] = 2;
+        fTest2[1][0] = 3;
+        fTest2[1][1] = 4;
+
+        for (int i=0;i<2;i++)
+          for (int j=0;j<3;j++)
+            for (int k=0;k<4;k++) {
+               fBool[i][j][k] = (i+j) % 2 !=0;
+               fChar[i][j][k] = 48 + i+j+k;
+               fShort[i][j][k] = i+j+k;
+               fInt[i][j][k] = i*j+k;
+               fLong[i][j][k] = i*j*k;
+               fFloat[i][j][k] = i*j*k;
+               fDouble[i][j][k] = i*j*k;
+            }
+     }
+
+     virtual ~TJsonEx2() {}
+
+     void Print()
+     {
+         for (int i=0;i<4;i++)
+           cout << "   fTest1[" << i << "] = " << fTest1[i] << endl;
+
+         for (int i=0;i<2;i++)
+         for (int j=0;j<3;j++)
+         for (int k=0;k<4;k++) {
+            cout << "  for indexes ["<<i<<"]["<<j<<"]["<<k<<"]" << endl;
+            cout << "     fBool = " << fBool[i][j][k] << endl;
+            cout << "     fChar = " << fChar[i][j][k] << endl;
+            cout << "     fShort = " << fShort[i][j][k] << endl;
+            cout << "     fInt = " << fInt[i][j][k] << endl;
+            cout << "     fLong = " << fLong[i][j][k] << endl;
+            cout << "     fFloat = " << fFloat[i][j][k] << endl;
+            cout << "     fDouble = " << fDouble[i][j][k] << endl;
+         }
+     }
+};
+
+// _______________________________________________________________
+
+class TJsonEx3 {
+   protected:
+     int        fSize;
+     bool       *fBool;     // [fSize]
+     char       *fChar;     // [fSize]
+     short      *fShort;    // [fSize]
+     int        *fInt;      // [fSize]
+     long       *fLong;     // [fSize]
+     float      *fFloat;    // [fSize]
+     double     *fDouble;   // [fSize]
+   public:
+
+     TJsonEx3()
+     {
+        fSize = 5;
+        fBool = new bool[fSize];
+        fChar = new char[fSize];
+        fShort = new short[fSize];
+        fInt = new int[fSize];
+        fLong = new long[fSize];
+        fFloat = new float[fSize];
+        fDouble = new double[fSize];
+
+        for (int n=0;n<fSize;n++) {
+           fBool[n] = false;
+           fChar[n] = 49 + n;
+           fShort[n] = n*2;
+           fInt[n] = n*5;
+           fLong[n] = n*123;
+           fFloat[n] = n*3;
+           fDouble[n] = n*7;
+        }
+
+     }
+     virtual ~TJsonEx3()
+     {
+       //delete[] fBool;
```

Comment:
This was due to missing copy constructor. 
I used this class in std::vector and arrays pointers were copied 
Now class should work correctly.